static/pages/redirect.html

```diff
@@ -0,0 +1,141 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <link href='http://fonts.googleapis.com/css?family=Roboto' rel='stylesheet' type='text/css'>
+    <style>
+
+        body {
+            font-family: 'Roboto', sans-serif;
+        }
+        .button {
+            color: white;
+            padding: 8px 16px;
+            text-align: center;
+            text-decoration: none;
+            font-size: 16px;
+            margin: 4px 2px;
+            transition-duration: 0.4s;
+            cursor: pointer;
+        }
+
+        .buttonReload {
+            background-color: white;
+            color: black;
+            border: 2px solid #e7e7e7;
+            border-radius: 8px;
+        }
+
+        .buttonReload:hover {
+            background-color: #e7e7e7;
+        }
+
+        .container {
+            display: flex;
+            align-items: center;
+            justify-content: center;
+            flex-direction: column;
+            height: 720px;
+        }
+
+        .item {
+            max-width: 75%;
+        }
+
+        details {
+            width: 400px;
+            border: 1px solid #aaa;
+            border-radius: 4px;
+            padding: .5em .5em 0;
+        }
+
+        .detailsContent {
+            margin: auto;
+            display: flex;
+            justify-content: center;
+            width: 100%;
+        }
+
+        summary {
+            text-align: center;
+            font-weight: bold;
+            margin: -.5em -.5em 0;
+            padding: .5em;
+        }
+
+
+
+
+
+    </style>
+</head>
+<body>
+
+<div class="container">
+
+    <div class="item"><h2>something goes wrong with this url:</h2></div>
+
+    <details class="item" id="errorDetails">
+        <summary div id="url"> </summary>
+        <div class="detailsContent">
+        <p id="error" style="color: rgb(255, 0, 0,0.8)"> </p>
+        </div>
+        <div class="detailsContent">
+            <button id="reload" class="button buttonReload" onclick="reload()"> reload </button>
+        </div>
+    </details>
+    <br>
+    <div id="timer" class="item"></div>
+</div>
+
+
+<script>
+    let {lastError, reloadTimeout,reloadInterval, url} = {}
```

Comment:
We do not need these globals. We can pass the necessary data as function's arguments, which is a lot cleaner. You have a clear scope on what your function needs and will be sure that it has as little interaction as possible outside of it.
Here we would only need Timer since it has to be used both by getDetails and runTimer